moped-editor/src/views/projects/projectView/KnackSync.js

```diff
@@ -0,0 +1,217 @@
+import React from "react";
+import { MenuItem, Icon, ListItemIcon, ListItemText } from "@material-ui/core";
+
+import { useMutation } from "@apollo/client";
+
+import { UPDATE_PROJECT_KNACK_ID } from "../../../queries/project";
+
+/**
+ * Entry in project menu drop down to trigger sync to Data Tracker
+ * @return {JSX.Element}
+ * @constructor
+ */
+const KnackSync = React.forwardRef(
+  (
+    {
+      project,
+      closeHandler,
+      snackbarHandler,
+      sceneNumber = 514,
+      viewNumber = 3047,
+      knackApplicationId = "6167314778435d001ea3e7cb",
+    },
+    ref
+  ) => {
+    /**
+     * Function to build the correct Knack URL to interact with based on properties and if there will be an
+     * update or an initial sync.
+     * @returns string
+     */
+    const buildUrl = () => {
+      let url =
+        "https://api.knack.com/v1/pages/scene_" +
```

Comment:
for string concats/substitutions, use [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals):
 ```javascript
`https://api.knack.com/v1/pages/scene_${sceneNumber}/views/${viewNumber}/records`
```

Reply:
i think your code will be cleaner if you just assign the url to a variable every render (or use a hook), rather than calling `buildUrl()` within your fetch.  ideally, you would hoist this function out of the component and use it like so:

so
```javascript
const url = buildUrl({scene, view, projectId});
```
i'm not very consistent when deciding when to hoist functions out of components, but the basic idea is that it is more performant, since the function itself is otherwise created on every render. performance won't matter here, but the code will look nicer.

Reply:
agree with John on both of these suggestions.

Reply:
These suggestions are helpful, thanks y'all. I have incorporated them.